### ISSUE TITLE
Revise License Server / Licensing Methodology

### ISSUE THREAD

**Issue by pacmano1** [nextgenhealthcare/connect#4471]
1. Allow offline activation for those behind restrictive firewalls (e.g. DoD) (e.g. Microsoft does this with Remote Desktop Licenses)
2. Allow self service of license keys at the success community to view, terminate and re-issue licenses.   Calling the support team is resource intensive on your side and time consuming on your clients' side.
3. Create API for automating license movement callable by your clients.
4. Licensing methodology not to great for containers (extra licenses is not the best solution).
5. Allow authenticated proxy for connection to license server.
6. Send warnings / countdowns for license renewal (e.g. 90, 60, 30, 7,6,5,4,3,2,1 days out, or configurable)
7. Deliver the license manager plugin as standard or put it in plug-in central on the success community.

**Comment by Jorgisimo** [comment]
yes! we are just moving our install to docker and the licensing has us scratching our head. we are doing a lot of testing so we are recreating a ton so we have a thread open with nextgen just asking to delete licenses daily.
